services/web/server/src/simcore_service_webserver/studies_access.py

```diff
@@ -55,41 +55,57 @@ async def get_public_project(app: web.Application, project_uuid: str):
     return prj
 
 
-# TODO: from .users import create_temporary_user
 async def create_temporary_user(request: web.Request):
     """
-        TODO: user should have an expiration date and limited persmissions!
+    TODO: user should have an expiration date and limited persmissions!
     """
     from .login.cfg import get_storage
-    from .login.handlers import ACTIVE, GUEST
+    from .login.handlers import ACTIVE, ANONYMOUS
     from .login.utils import get_client_ip, get_random_string
     from .security_api import encrypt_password
 
-    # from .utils import generate_passphrase
-    # from .utils import generate_password
-
     db = get_storage(request.app)
 
     # TODO: avatar is an icon of the hero!
-    # FIXME: # username = generate_passphrase(number_of_words=2).replace(" ", "_").replace("'", "")
     username = get_random_string(min_len=5)
     email = username + "@guest-at-osparc.io"
     password = get_random_string(min_len=12)
 
+    # creates a user that is marked as ANONYMOUS. see update_user_as_guest
     user = await db.create_user(
         {
             "name": username,
             "email": email,
             "password_hash": encrypt_password(password),
             "status": ACTIVE,
-            "role": GUEST,
+            "role": ANONYMOUS,
             "created_ip": get_client_ip(request),
         }
     )
 
     return user
 
 
+async def activate_as_guest_user(user: Dict, app: web.Application):
+    from .login.cfg import get_storage
+    from .login.handlers import GUEST, ANONYMOUS
+    from .resource_manager.websocket_manager import WebsocketRegistry
+
+    if user.get("role") == ANONYMOUS:
+        db = get_storage(app)
+        username = user["name"]
+
+        # creates an entry in the socket's registry to avoid garbage collector (GC) deleting it
+        # See https://github.com/ITISFoundation/osparc-simcore/issues/1853
+        registry = WebsocketRegistry(user["id"], f"{username}-guest-session", app)
+        await registry.set_socket_id(f"{username}-guest-socket-id")
```

Comment:
So this will give you exactly 1 period of time, equal to the value of `resource_deletion_timeout_seconds` which defaults to 900 seconds (if was not changes on the platforms). In this period of time the user will not be removed by the GC.

This will give `resource_deletion_timeout_seconds ` seconds for the user to be logged in. Please keep in mind that there is a mechanism where if the user pressed logout, its resources will be collected even faster. Search for `# trigger faster gc on disconnect`.

If I remember correctly e2e also presses logout when it finishes with a test.

Reply:
> So this will give you exactly 1 period of time, equal to the value of `resource_deletion_timeout_seconds` which defaults to 900 seconds (if was not changes on the platforms). In this period of time the user will not be removed by the GC.

Correct. In any case, if you look in detail, in reality resource_deletion_timeout_seconds will be at least 3 secs. I can

> This will give `resource_deletion_timeout_seconds ` seconds for the user to be logged in.

Yes. But notice that that value starts counting from the moment the server responds **with a redirection** ... meaning that it should immediately 


 > Please keep in mind that there is a mechanism where if the user pressed logout, its resources will be collected even faster. Search for `# trigger faster gc on disconnect`.

That is fine. A guest user that logs out should be GC-ed